### DISCUSSION TITLE
How to generate reports using `@siteimprove/alfa-cypress`

### DISCUSSION THREAD

**Discussion by yoshi-doo** [Siteimprove/alfa#1701] [Questions] [answered]
Hey,

I am trying out  `@siteimprove/alfa-cypress` for a project, I ran the example from https://github.com/Siteimprove/alfa-examples/tree/main/end-to-end-testing/cypress and it produces a json file (`page.spec.json`) with content  `[object Object]`,
How am i supposed to use this ? my end goal is to extract a report and upload it siteimprove dashboard.

Thanks !

**Comment by Jym77** [comment, marked as answer]
Hi,
yes, this is actually broken and poorly serialized 🙈 (i.e., this is Javascript calling `toString` on a object and it's too late to unpack it). If I remember correctly, it's [the persister](https://github.com/Siteimprove/alfa-examples/blob/main/common/persist.ts) that is missing an `await` somewhere.

We are currently working on a better Cypress integration together with the [alfa-test-utils package](https://github.com/Siteimprove/alfa-integrations/tree/main/packages/alfa-test-utils) ([temporary documentation](https://alfa.siteimprove.com/code-checker)) (the Cypress integration is on its way, current ETA end of October). Hopefully, that will work for you 🤞

**Reply by yoshi-doo** [reply]
Hi,

Thank you for the quick response and for confirming the issue.

The upcoming Cypress integration with the alfa-test-utils package sounds promising! I'll look forward to the October release and hope it resolves our current challenges.

Thanks again for the update and all the work going into this.

**Comment by Jym77** [comment]
Hi @yoshi-doo 
The Cypress example is now ready 😊 You can see it [in the Alfa examples repository](https://github.com/Siteimprove/alfa-examples/tree/main/accessibility-code-checker/cypress). It's meant to be cloned and experimented with locally, so go for it!

The documentation [in the Alfa hub](https://alfa.siteimprove.com/code-checker) is on its way, probably for the end of the week.

I'll close this discussion, feel free to reopen if you have more questions.

**Reply by yoshi-doo** [reply]
Thanks a lot for the update :)
